coordinator/app/src/main/kotlin/net/consensys/zkevm/coordinator/app/LastFinalizedBlockProvider.kt

```diff
@@ -36,7 +36,13 @@ class L1BasedLastFinalizedBlockProvider(
   override fun getLastFinalizedBlock(): SafeFuture<ULong> {
     lineaRollupSmartContractWeb3jClient.setDefaultBlockParameter(DefaultBlockParameterName.LATEST)
 
-    return SafeFuture.of(lineaRollupSmartContractWeb3jClient.currentL2BlockNumber().sendAsync())
+    return AsyncRetryer.retry(
+      vertx,
+      maxRetries = numberOfRetries.toInt(),
+      backoffDelay = pollingInterval
+    ) {
+      SafeFuture.of(lineaRollupSmartContractWeb3jClient.currentL2BlockNumber().sendAsync())
+    }
```

Comment:
Add retry here to wait until `currentL2BlockNumber` on L1 is present, otherwise will get the following error:
```
time=2024-12-12T06:25:21,230Z level=FATAL message=org.web3j.tx.exceptions.ContractCallException: Empty value (0x) returned from contract | logger=CoordinatorAppCli thread=main | java.util.concurrent.ExecutionException: org.web3j.tx.exceptions.ContractCallException: Empty value (0x) returned from contract
2024-12-12 14:25:21     at java.base/java.util.concurrent.CompletableFuture.reportGet(Unknown Source)
2024-12-12 14:25:21     at java.base/java.util.concurrent.CompletableFuture.get(Unknown Source)
2024-12-12 14:25:21     at net.consensys.zkevm.coordinator.app.L1DependentApp.<init>(L1DependentApp.kt:308)
...
```